code/espurna/sensor.ino

```diff
@@ -234,73 +231,79 @@ void _sensorWebSocketSendData(JsonObject& root) {
+            if (sns->getID() == SENSOR_HLW8012_ID) {
+                sensor["hlwVisible"] = 1;
+                sensor["pwrVisible"] = 1;
             }
         #endif
 
         #if CSE7766_SUPPORT
-            if (sensor->getID() == SENSOR_CSE7766_ID) {
-                root["cseVisible"] = 1;
-                root["pwrVisible"] = 1;
+            if (sns->getID() == SENSOR_CSE7766_ID) {
+                sensor["cseVisible"] = 1;
+                sensor["pwrVisible"] = 1;
             }
         #endif
 
         #if V9261F_SUPPORT
-            if (sensor->getID() == SENSOR_V9261F_ID) {
-                root["pwrVisible"] = 1;
+            if (sns->getID() == SENSOR_V9261F_ID) {
+                sensor["pwrVisible"] = 1;
             }
         #endif
 
         #if ECH1560_SUPPORT
-            if (sensor->getID() == SENSOR_ECH1560_ID) {
-                root["pwrVisible"] = 1;
+            if (sns->getID() == SENSOR_ECH1560_ID) {
+                sensor["pwrVisible"] = 1;
             }
         #endif
 
         #if PZEM004T_SUPPORT
-            if (sensor->getID() == SENSOR_PZEM004T_ID) {
-                root["pzemVisible"] = 1;
-                root["pwrVisible"] = 1;
+            if (sns->getID() == SENSOR_PZEM004T_ID) {
+                sensor["pzemVisible"] = 1;
+                sensor["pwrVisible"] = 1;
             }
         #endif
 
         #if PULSEMETER_SUPPORT
-            if (sensor->getID() == SENSOR_PULSEMETER_ID) {
-                root["pmVisible"] = 1;
-                root["pwrRatioE"] = ((PulseMeterSensor *) sensor)->getEnergyRatio();
+            if (sns->getID() == SENSOR_PULSEMETER_ID) {
+                sensor["pmVisible"] = 1;
+                sensor["pwrRatioE"] = ((PulseMeterSensor *) sns)->getEnergyRatio();
             }
         #endif
 
     }
 
-    if (magnitudeCount()) {
-        //root["apiRealTime"] = _sensor_realtime;
-        root["pwrUnits"] = _sensor_power_units;
-        root["eneUnits"] = _sensor_energy_units;
-        root["tmpUnits"] = _sensor_temperature_units;
-        root["tmpCorrection"] = _sensor_temperature_correction;
-        root["humCorrection"] = _sensor_humidity_correction;
-        root["snsRead"] = _sensor_read_interval / 1000;
-        root["snsReport"] = _sensor_report_every;
-        root["snsSave"] = _sensor_save_every;
-        _sensorWebSocketMagnitudesConfig(root);
-    }
+
+    JsonObject& sensor = root.createNestedObject("sensor");
+
+    _sensorWebSocketMagnitudesConfig(sensor);
+        
+    //TODO why is this not nested?
+    sensor["pwrUnits"] = _sensor_power_units;
+    sensor["eneUnits"] = _sensor_energy_units;
+    sensor["tmpUnits"] = _sensor_temperature_units;
+    sensor["tmpCorrection"] = _sensor_temperature_correction;
+    sensor["humCorrection"] = _sensor_humidity_correction;
+    sensor["snsRead"] = _sensor_read_interval / 1000;
+    sensor["snsReport"] = _sensor_report_every;
+    sensor["snsSave"] = _sensor_save_every;
```

Comment:
In this module, what are all this values and why are they not per magnitude?

Reply:
Correction code makes more sense to have per sensor instead of globally, perhaps the assumption was that the device only uses a single sensor or sensors are all in the same place / have similar problems that require correction

Read / report values could go per sensor too, since those could have a different requirements. idk about saving, because the only use case right now is energy (which, btw, is saved per sensor as eneTotalN)

Units could be kept global? Is there a reason to have them specialized?